simplesecurity/__init__.py

```diff
@@ -1,212 +1,285 @@
-"""
-Combine multiple popular python security tools and generate reports or output
-into different formats
-
-Plugins (these require the plugin executable in the system path. e.g. bandit
-requires bandit to be in the system path...)
-
-- bandit
-- safety
-- dodgy
-- dlint
-- semgrep
-
-Formats
-
-- ansi (for terminal)
-- json
-- markdown
-- csv
-- sarif
-"""
 from __future__ import annotations
 
 import argparse
+import logging
+import os
 from sys import exit as sysexit
 from sys import stdout
 from typing import Any
 
 from simplesecurity import filter as secfilter
 from simplesecurity import formatter, plugins
+from simplesecurity.github import GithubAnnotationsAndComments
 from simplesecurity.types import Finding
 
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s %(name)-6s %(levelname)-6s %(message)s",
+    datefmt="%m-%d %H:%M",
+    handlers=[logging.StreamHandler()],
+)
+
+logger = logging.getLogger()
+
 stdout.reconfigure(encoding="utf-8")  # type:ignore
 FORMAT_HELP = "Output format. One of ansi, json, markdown, csv. default=ansi"
-PLUGIN_HELP = "Plugin to use. One of bandit, safety, dodgy, dlint, semgrep, all, default=all"
+PLUGIN_HELP = "Plugin to use. One of bandit, safety, dodgy, dlint, semgrep, trivy or all, default=all"
+SCAN_PATH = (
+    "Define Path that should be scannend, default path is root of CLI tool"
+)
 
 
 def runAllPlugins(
-	pluginMap: dict[str, Any], severity: int, confidence: int, fast: bool
+    scan_path: str,
+    pluginMap: dict[str, Any],
+    severity: int,
+    confidence: int,
+    fast: bool,
 ) -> list[Finding]:
-	"""Run each plugin. Optimise as much as we can.
-
-	Args:
-		pluginMap (dict[str, Any]): the plugin map
-		severity (int): the minimum severity to report on
-		confidence (int): the minimum confidence to report on
-		fast (bool): runAllPlugins with optimisations
-
-	Returns:
-		list[Finding]: list of findings
-	"""
-	findings: list[Finding] = []
-	for plugin in pluginMap:
-		# Do optimisations
-		if (
-			pluginMap[plugin]["max_severity"] >= severity
-			and pluginMap[plugin]["max_confidence"] >= confidence
-			and (not fast or pluginMap[plugin]["fast"])
-		):
-			try:
-				findings.extend(pluginMap[plugin]["func"]())
-			except RuntimeError as error:
-				print(error)
-	return findings
+    """This helper function triggers als scans if no specific scan is
+    requested. It triggers the scans chronologically.
+
+    :param scan_path: The scanning path is a string that point to the directory that should be scanned. This argument is required.
```

Comment:
Could we have typing in docstrings please? Also I think this still uses spaces atm :)

Reply:
Everything is still using spaces. I am looking into how to revert this to tabs. The typing is included in the sphinx documentation, as this is directly imported from the function definitions by the sphinx-autodoc-typehints module of sphinx. This should make coding and documentation more consistent.